src/plugins/data/public/query/query_string/language_service/lib/recent_query.tsx

```diff
@@ -51,7 +51,10 @@ export function RecentQueriesTable({
           icon: 'play',
           type: 'icon',
           onClick: (item: RecentQueryTableItem) => {
-            onClickRecentQuery(recentQueries[item.id].query, recentQueries[item.id].timeRange);
+            onClickRecentQuery(
```

Comment:
to mind explaining this part a little bit? like how this fixes the problem?

without knowing too much about the implementation, for me this basically looks for the query and assumes its the same query which might not be accurate still i think.

i dont think we show this feature for DQL but in the case of DQL the query could actually just be 
`""` for multiple queries for different index patterns. And this logic to me would find the first query that is `""` and assume that is the recent query.

in SQL and PPL we dont limit what they type so technically they could select `data_logs`

and in SQL it would be `SELECT * FROM data_logs`
and then they type it `SELECT * FROM eccomerce` and then select the eccommerce sample data. so then the query editor says `SELECT * FROM eccomerce` which query is the right query here.

i think the solution here should be assigning a GUID or using the timestamp of execution as the identifier

Reply:
make sense. i revised it to use uuid